app/views/posts/show.html.erb

```diff
@@ -0,0 +1,26 @@
+<p id="notice"><%= notice %></p>
+<div class="card my-3 col-6 m-auto">
+  <div class="card-body">
+    <h1 class="card-title">
+      <%= @post.title %>
+      
+      <% if user_signed_in? %>
+        <small class="text-muted"><small><em>by
+        <%= @post.user.username %></em></small></small>
+      <% else %>
+        <small class="text-muted"><small><em>by
+        Anonymous</em></small></small>
+      <% end %>
```

Comment:
View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)